### PR TITLE
provider/cloudflare: Add migration for v1 to v4 API libraries

### LINKED ISSUE #6620: Cloudflare Record Fails on Existing Resources in 0.6.16
### Terraform Version
0.6.16

### Affected Resource(s)
- cloudflare_record

### Terraform Configuration Files
```hcl
resource "aws_instance" "vpn" {
  ami = "[REDACTED]"
  instance_type = "t2.micro"
  subnet_id = "[REDACTED]"
  security_groups = ["[REDACTED]"]
  key_name = "vpn-key"
  source_dest_check = false
}

resource "aws_eip" "vpn" {
    instance = "${aws_instance.vpn.id}"
    vpc = true
    depends_on = ["aws_instance.vpn"]
}

resource "cloudflare_record" "vpn" {
    domain = "[REDACTED]"
    name = "vpn"
    value = "${aws_eip.vpn.public_dns}"
    type = "CNAME"
    ttl = 1
}
```

### Expected Behavior
I was building only to replace an EC2 instance with a new one running a new AMI. I expected the instance to get replaced and assigned the Elastic IP assigned to it. The Cloudflare record should not have been modified since it's tied to the EIP. I expected the build to just replace the EC2 instance like it would have in 0.6.15

### Actual Behavior
error:
* cloudflare_record.vpn: Error from makeRequest: {"success":false,"errors":[{"code":7003,"message":"Could not route to \/zones\/=[REDACTED]\/dns_records\/REDACTED], perhaps your object identifier is invalid?"},{"code":7000,"message":"No route for that URI"}],"messages":[],"result":null}

### Steps to Reproduce
1. run terraform plan or apply on a pre-existing Terrform project with a Cloudflare Record. You do not need to have changed the cloudflare_record resource.

The Cloudflare record nor the zone have been modified outside of Terrafom. I am able to reproduce this in every Terraform project I have with a cloudflare_record resource. So it is not limited to a particular instance. All resources work correctly if I roll back to Terraform 0.6.15.

Perhaps this relates somehow to #5508 or #6449 as those look to be the only changes to the cloudflare_record resource in 0.6.16?

### PR DESCRIPTION
**Pull request by catsby**
Fixes #6620 by providing a migration from records created with the old library and v1 of the API, to the new library with v4. Unfortunately we have no way of doing this on CloudFlare's side (can't query for the new version of a given ID), so we have to manually match against the records. 

TODO:

- [x] Upgrade internal library to the latest upstream (https://github.com/hashicorp/terraform/pull/7037)
- [x] Rebase this PR after ^ gets merged
- [x] Stub out some cloud flare server responses for a test here
- [x] 💸

### CODE REVIEW

File: builtin/providers/cloudflare/resource_cloudflare_record_migrate.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/terraform"
+	// NOTE: Temporary until they merge my PR:
+	"github.com/mitchellh/cloudflare-go"
+)
+
+func resourceAwsCloudFlareRecordMigrateState(
```
Comment (kyl191) [inline builtin/providers/cloudflare/resource_cloudflare_record_migrate.go]: Small nit: Cloudflare isn't part of AWS, so I don't think there's a reason to include 'AWS' in the function name.

Comment (catsby) [inline builtin/providers/cloudflare/resource_cloudflare_record_migrate.go]: 😱 thanks :)

**Comment by phinze** [comment]
Code LGTM - is there a way to wrap a test or three around this?

File: builtin/providers/cloudflare/resource_cloudflare_record_migrate_test.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+				"name":     "hashicorptest.com",
+				"content":  "some.registrar-servers.com",
+				"ttl":      "1",
+				"priority": "10",
+				"zone_id":  "1234567890",
+				"domain":   "hashicorptest.com",
+			},
+			Expected:   "12342092cbc4c391be33ce548713bba3",
+			ShouldFail: true,
+		},
```
Comment (phinze) [inline builtin/providers/cloudflare/resource_cloudflare_record_migrate_test.go]: Nice!

Looks like these exercise TTL and Priority matching - need another pair for exercising the "proxied" logic too?

**Comment by phinze** [comment]
One test Q and one cranky `go vet` to fix - nearly ready to go! 🚀

**Comment by catsby** [comment]
Sorry for the run-around; I added a test case for [not]proxied and fixed the `go vet` thing

**Comment by phinze** [comment]
👍  LGTM pending travis

**Pull request merged**